### PR TITLE
Include "Product Name" in status.php printout

### PR DESCRIPTION
**Pull request by mmattel**
Refs: #25822 [Feature Request] status.php should also print out the source name

After discussing I am not using the vendor name but the product name due to branding possibilities.
The product name is either the default (ownCloud) or any defined one by the standard branding procedure.

**New output:**
`{"installed":true,"maintenance":false,"version":"9.2.0.1","versionstring":"9.2.0 pre alpha","edition":"","productname":"ownCloud"}`

@PVince81 
(in case of, support for backporting needed...)

### CODE REVIEW

File: status.php [hunk 1]
```diff
@@ -34,12 +34,18 @@
 
 	$installed = (bool) $systemConfig->getValue('installed', false);
 	$maintenance = (bool) $systemConfig->getValue('maintenance', false);
+	$defaults = new \OC_Defaults();
+	$defaults->getName();
```
Comment (PVince81) [inline status.php]: no need for this, it was only the example

File: status.php [hunk 2]
```diff
@@ -34,12 +34,18 @@
...
+	# see core/lib/private/legacy/defaults.php and core/themes/example/defaults.php
+	# for description and defaults
+	$identifyer = $defaults->getName();
```
Comment (PVince81) [inline status.php]: => "identifier", no "y" :smile:

**Comment by PVince81** [comment]
The code looks good. Would like to know what @DeepDiver1975 and @butonic think performance-wise.

File: status.php (hunk 2 above)
Comment (mmattel) [inline status.php]: fixed typo :smirk:

File: status.php (hunk 1 above)
Comment (mmattel) [inline status.php]: I went thru the code and tried to look for examples how to access this directly but did not found one. All of them are using ``new \OC_Defaults``... Pls excuse for asking but how is the alternative looking like?

File: status.php [hunk 3]
```diff
@@ -34,12 +34,18 @@
 
 	$installed = (bool) $systemConfig->getValue('installed', false);
 	$maintenance = (bool) $systemConfig->getValue('maintenance', false);
+	# see core/lib/private/legacy/defaults.php and core/themes/example/defaults.php
+	# for description and defaults
+	$defaults = new \OC_Defaults();
+	$defaults->getName();
```
Comment (PVince81) [inline status.php]: remove this line, you don't need it as you already call it below

Comment (mmattel) [inline status.php]: Done

File: status.php [hunk 4]
```diff
@@ -34,12 +34,16 @@
 
 	$installed = (bool) $systemConfig->getValue('installed', false);
 	$maintenance = (bool) $systemConfig->getValue('maintenance', false);
+	# see core/lib/private/legacy/defaults.php and core/themes/example/defaults.php
+	# for description and defaults
+	$defaults = new \OC_Defaults();
```
Comment (butonic) [inline status.php]: I admit we use `\OC_Defaults` nearly everywhere, when the rigth thing to do is use `\OCP\Defaults`. Please use that one.

**Comment by butonic** [comment]
Performance wise the theming logic (`require_once` a theme) should be cached by phps opcache. A Session is also already initialized. Impact should be minimal.

**Comment by butonic** [comment]
Lets get this over with. 👍

File: status.php (hunk 4 above)
Comment (mmattel) [inline status.php]: Not related to this PR but...
from what I have seen these are the files that would require an update from ``\OC_Defaults`` to ``\OCP\Defaults``. Shall I do PR for that?:

https://github.com/owncloud/core/blob/5a058f6c0ef64ff0272fcdb58478f9c6611d6819/apps/federation/lib/DAV/FedAuth.php#L41
https://github.com/owncloud/core/blob/5a058f6c0ef64ff0272fcdb58478f9c6611d6819/lib/private/Console/Application.php#L54
https://github.com/owncloud/core/blob/5a058f6c0ef64ff0272fcdb58478f9c6611d6819/core/js/config.php#L46
https://github.com/owncloud/core/blob/5a058f6c0ef64ff0272fcdb58478f9c6611d6819/apps/dav/lib/Connector/PublicAuth.php#L68
https://github.com/owncloud/core/blob/5a058f6c0ef64ff0272fcdb58478f9c6611d6819/core/Application.php#L182
https://github.com/owncloud/core/blob/5a058f6c0ef64ff0272fcdb58478f9c6611d6819/lib/private/URLGenerator.php#L225
https://github.com/owncloud/core/blob/1ab7ee5e231b85ffdb24c284e57cebdd030b8a3d/tests/lib/TestCase.php#L434
https://github.com/owncloud/core/blob/5a058f6c0ef64ff0272fcdb58478f9c6611d6819/apps/dav/lib/Connector/Sabre/Auth.php#L81
https://github.com/owncloud/core/blob/5a058f6c0ef64ff0272fcdb58478f9c6611d6819/core/Command/Maintenance/Install.php#L67
https://github.com/owncloud/core/blob/5a058f6c0ef64ff0272fcdb58478f9c6611d6819/settings/Application.php#L230
https://github.com/owncloud/core/blob/53f492d71de460f691b93ca3cc1cd2356f103d04/settings/personal.php#L40
https://github.com/owncloud/core/blob/5a058f6c0ef64ff0272fcdb58478f9c6611d6819/lib/private/legacy/template.php#L83
https://github.com/owncloud/core/blob/5a058f6c0ef64ff0272fcdb58478f9c6611d6819/lib/base.php#L827
https://github.com/owncloud/core/blob/5a058f6c0ef64ff0272fcdb58478f9c6611d6819/lib/private/Server.php#L564
https://github.com/owncloud/core/blob/c72f8594fed552f74d3ce075d914ccfc404a0134/lib/private/legacy/util.php#L638

Comment (butonic) [inline status.php]: that would be awesome!

**Comment by guruz** [comment]
👍

**Comment by mmattel** [comment]
Why did Mr. Jenkins fail?

**Comment by PVince81** [comment]
@mmattel because Mr Jenkins looked into the Oracle and was engulfed.

**Pull request merged**